internal/graph/check.go

```diff
@@ -703,35 +740,15 @@ func (c *LocalChecker) buildCheckAssociatedObjects(req *ResolveCheckRequest, obj
 		reqTupleKey := req.GetTupleKey()
 		reqContext := req.GetContext()
 
-		objectType, relation := tuple.SplitObjectRelation(objectRel)
-
-		user := reqTupleKey.GetUser()
-		userType := tuple.GetType(user)
-
-		relationReference := typesystem.DirectRelationReference(objectType, relation)
-		hasPubliclyAssignedType, _ := typesys.IsPubliclyAssignable(relationReference, userType)
-
-		userFilter := []*openfgav1.ObjectRelation{{
-			Object: user,
-		}}
-		if hasPubliclyAssignedType {
-			userFilter = append(userFilter, &openfgav1.ObjectRelation{
-				Object: tuple.TypedPublicWildcard(userType),
-			})
-		}
-
 		opts := storage.ReadStartingWithUserOptions{
 			Consistency: storage.ConsistencyOptions{
 				Preference: req.GetConsistency(),
 			},
 		}
 
-		i, err := ds.ReadStartingWithUser(ctx, storeID, storage.ReadStartingWithUserFilter{
-			ObjectType: objectType,
-			Relation:   relation,
-			UserFilter: userFilter,
-			ObjectIDs:  objectIDs,
-		}, opts)
+		// TODO: add in optimization to filter out user not matching the type
```

Comment:
Seems to me that the TODO lets us know that the typesystem bit should not be hidden down in the `userFilter` function as it would basically determine the type of iterator it would yield.

Reply:
Update function signature to take out typesystem from user filter